### PR TITLE
Revert previous package consolidation

### PR DESCRIPTION
**Pull request by dantleech**
As detailed in the ADR, the bundling together of packages is not practical.

Probably the only way to solve the packaging issues is the monorepo.

### CODE REVIEW

File: doc/adr/0001-package-structure-macro-packages.md [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+```
+Phpactor\\Extension\\<subject>Extension\\
+Phpactor\\Extension\\<subject>LanguageServerExtension\\
+Phpactor\\Extension\\<subject>RpcExtension\\
+Phpactor\\<subject>\\{Adapter|Bridge}\\
+Phpactor\\<subject>\\{Model|Core}\\
+```
+
+- The `Extension` connects the `<subject>` `Model` to the Phpactor
+  conatiner. There may be an Extension which is "pure" (abstract) and provide
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:20]:
```suggestion
  container. There may be an Extension which is "pure" (abstract) and provide
```

File: doc/adr/0001-package-structure-macro-packages.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  an exnteion point. Others will use the extension point to integrate
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:21]:
```suggestion
  an extension point. Others will use the extension point to integrate
```

File: doc/adr/0001-package-structure-macro-packages.md [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+  adapters (concrete implementations).
+- The `LanguageServer` and (Phpactor) `Rpc` extensions proide handlers for
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:23]:
```suggestion
- The `LanguageServer` and (Phpactor) `Rpc` extensions provide handlers for
```

File: doc/adr/0001-package-structure-macro-packages.md [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  these two RPC methods.
+- The adapters are the concrete implementations (e.g. `WorseReflection` for
+  fulfilling the completion APIs).
+- The `Model` is the pure domain code and APIs.
+
+Each separate namespace as a special concern. Originally it was intended that
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:29]:
```suggestion
Each separate namespace has a special concern. Originally it was intended that
```

File: doc/adr/0001-package-structure-macro-packages.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+package stability principles be followed and each package would live as a
+seperate repository with a separate package (let's call them micro-packages).
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:31]:
```suggestion
separate repository with a separate package (let's call them micro-packages).
```

File: doc/adr/0001-package-structure-macro-packages.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+While we could have provided stubs in the old namespace in the packages, many
+packages exposed multiple public classes, so it wasn't practical.
+
+We solved this by mapping `Phpactor\\` to `lib/` and move everything in to keep the
+same namespace structure as before. (e.g. `Phpactor\\LanguageServer``,
+`Phpactor\\Extension\\LanguageServer`>
+
+But finally the **whole idea is flawed**: The "extension" dependencies were in
+`require-dev`, which meant that packages depending on an extension, would need
+to explicitly require the package, the extensions and any other depdencies of
```
Comment (greg0ire) [inline doc/adr/0001-package-structure-macro-packages.md:103]:
```suggestion
to explicitly require the package, the extensions and any other dependencies of
```

**Comment by dantleech** [comment]
thanks @greg0ire , i did `:set spell` and fixed the errors :)

File: lib/Phpactor.php [hunk 7]
```diff
@@ -38,12 +39,11 @@
...
 use Phpactor\Extension\ReferenceFinder\ReferenceFinderExtension;
 
-class_alias(CompletionExtension::class, 'Phpactor\Extension\Completion\CompletionExtension');
-
 class Phpactor
 {
     private const LEGACY_EXTENSIONS = [
-        '\Phpactor\Extension\LanguageServerCompletion\LanguageServerCompletionExtension'
+        '\Phpactor\Extension\LanguageServerCompletion\LanguageServerCompletionExtension',
+        '\Phpactor\Extension\LanguageServer\LanguageServerExtension'
```
Comment (dantleech) [inline lib/Phpactor.php:46]: this no longer makes any difference

**Comment by dantleech** [comment]
cc @przepompownia

File: doc/adr/0001-package-structure-macro-packages.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+Phpactor\\Extension\\<subject>RpcExtension\\
+Phpactor\\<subject>\\{Adapter|Bridge}\\
+Phpactor\\<subject>\\{Model|Core}\\
+```
+
+- The `Extension` connects the `<subject>` `Model` to the Phpactor
+  container. There may be an Extension which is "pure" (abstract) and provide
+  an extension point. Others will use the extension point to integrate
+  adapters (concrete implementations).
+- The `LanguageServer` and (Phpactor) `Rpc` extensions pride handlers for
```
Comment (przepompownia) [inline doc/adr/0001-package-structure-macro-packages.md:23]: Probably there is something not detected by spell checker ;)

File: doc/adr/0001-package-structure-macro-packages.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+  these two RPC methods.
+- The adapters are the concrete implementations (e.g. `WorseReflection` for
+  fulfilling the completion APIs).
+- The `Model` is the pure domain code and APIs.
+
+Each separate namespace as a special concern. Originally it was intended that
```
Comment (przepompownia) [inline doc/adr/0001-package-structure-macro-packages.md:29]: "has"? (my English is poor as you see).

**Comment by przepompownia** [comment]
Independently of your decision, its explanation gives me an opportunity to looking at the whole structure. 

Jumping to definition and hover are still unavailable in bundled version. 
```
phpactor/language-server-reference-finder-extension
phpactor/language-server-hover-extension
```

**Comment by przepompownia** [comment]
@dantleech did you test what any user (or any post-install script) have to do after installation if given extension exist both in `extensions/phpactor/` and `vendor/phpactor/`?. 

At the moment the easier solution to completely remove `vendor` and `extensions` directories and refresh it again. Uninstalling by command does not work
```

 $ ./bin/phpactor ex:rem phpactor/language-server-completion-extension
Removing: phpactor/language-server-completion-extension

In RemoverService.php line 61:
                                                          
  Extension is a primary extension and cannot be removed  
                                                          

extension:remove <extension>...
```
so it could be probably `pre-install-cmd` composer script.

**Comment by dantleech** [comment]
> so it could be probably pre-install-cmd composer script.

that's a good idea. As far as I know not many people are using the language server extensions though, so the disruption should be minimal :crossed_fingers:

**Comment by dantleech** [comment]
(thanks for reading the ADR, I updated the mistakes)

**Pull request merged**